From bbd69e321f4f141900cf2184e109a8b144bf690a Mon Sep 17 00:00:00 2001
From: Ren Chenglei <chenglei.ren@intel.com>
Date: Fri, 9 Jun 2023 21:32:45 +0530
Subject: [PATCH] INTERNAL: Fix HWC crash issue when playing video on GVT-d

Add YUYV format for 264 video which is used by HWC.

Tracked-On: OAM-95589
Signed-off-by: Ren Chenglei <chenglei.ren@intel.com>
---
 i915.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/i915.c b/i915.c
index dabdced..d9da277 100644
--- a/i915.c
+++ b/i915.c
@@ -36,7 +36,12 @@ static const uint32_t scanout_render_formats[] = { DRM_FORMAT_ABGR2101010, DRM_F
 static const uint32_t render_formats[] = { DRM_FORMAT_ABGR16161616F };
 
 static const uint32_t texture_only_formats[] = { DRM_FORMAT_R8, DRM_FORMAT_NV12, DRM_FORMAT_P010,
+#ifdef USE_GRALLOC1
+						 DRM_FORMAT_YVU420, DRM_FORMAT_YVU420_ANDROID,
+						 DRM_FORMAT_YUYV };
+#else
 						 DRM_FORMAT_YVU420, DRM_FORMAT_YVU420_ANDROID };
+#endif
 
 static const uint64_t gen_modifier_order[] = { I915_FORMAT_MOD_Y_TILED_CCS, I915_FORMAT_MOD_Y_TILED,
 					       I915_FORMAT_MOD_X_TILED, DRM_FORMAT_MOD_LINEAR };
-- 
2.40.1

